deps/rabbit/src/rabbit_stream_coordinator.erl

```diff
@@ -425,7 +425,7 @@ ensure_coordinator_started() ->
     end.
 
 start_coordinator_cluster() ->
-    Nodes = rabbit_nodes:list_running(),
+    Nodes = rabbit_nodes:list_reachable(),
     rabbit_log:debug("Starting stream coordinator on nodes: ~w", [Nodes]),
```

Comment:
What about adding an assert like:
```erlang
Nodes = rabbit_nodes:list_reachable(),
rabbit_log:debug("Starting stream coordinator on nodes: ~w", [Nodes]),
?assertNotEqual([], Nodes),
...
```
... to catch an unwanted empty list early?

Reply:
You will have to include `assert.hrl` in this case:
```erlang
-include_lib("stdlib/include/assert.hrl").
```

Reply:
Not a fan of such asserts, especially in code that isn't just tests. The rabbit_log:debug will already tell us there are no nodes. So just doing `true = Nodes =/= []` should be enough.

Reply:
I prefer the assertion macros because you get more details about the context (not only a vague badmatch exception) and the intent is clearer. But that's personal preferences, not a rule :-)

Reply:
Crash reports have a lot more info these days and the info you lack is in the log just above on line 429. Even if you didn't have that log, looking at the crash line would be enough in this case.

There aren't many cases where the macros provide more info now, and when they do it's easy to have the same without the macro (for example `Expected = Result` could be written `{Expected, Expected} = {Result, Expected}`). But those cases are rare outside of tests.